### DISCUSSION TITLE
naming strategy for version name in pubspec.yaml for iOS

### DISCUSSION THREAD

**Discussion by elemental21** [codemagic-ci-cd/codemagic-docs#1685] [Q&A]
Hi, we are implementing a naming strategy where our staging builds get a -staging as part of the build number to differentiate from App Tester. Works fine for Android but on iOS it strips the “-staging” out so we can't tell which build is on the staging environment and which is on the production environment. Anyone know why or what we can do to differentiate the staging builds? We've done the even/odd thing before and I suppose we could do it but it's not ideal for our process.
`version: 0.44.0-staging`

so in android you’ll see 0.44.0-staging (899) but the iOS side says 0.44.0 (899)

**Comment by icarusdust** [comment]
Hi @elemental21, what have you tried exactly, have you tried adding -staging to `--build-number `? like the following one:
 
`--build-number=$PROJECT_BUILD_NUMBER-staging`

**Comment by vbuberen** [comment]
On iOS and MacOS only digits and dot symbols are allowed to be used in version. We even had to mention it in `known issues` in `package_info_plus` repo: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/package_info_plus/package_info_plus#plugin-returns-incorrect-app-version

**Reply by icarusdust** [reply]
Thanks for the update @vbuberen. As this looks to be an issue on their end, I'll close the thread. Please let us know in case of any further questions.
